internal/core/item_tree.rs

```diff
@@ -102,6 +104,82 @@ impl ItemTreeNode {
     }
 }
 
+/// The `ComponentItemTree` provides tree walking code for the physical ItemTree stored in
+/// a `Component` without stitching any inter-Component links together!
+pub struct ComponentItemTree<'a> {
+    item_tree: &'a [ItemTreeNode],
+}
+
+impl<'a> ComponentItemTree<'a> {
+    /// Create a new `ItemTree` from its raw data.
+    pub fn new(item_tree: &'a [ItemTreeNode]) -> Self {
```

Comment:
All of the call sites pass `comp_ref_pin.as_ref().get_item_tree().as_slice()`. It think the function should take a component ref pin as argument.